client/src/helpers/index.js

```diff
@@ -238,3 +259,215 @@ export const getDayTimeNow = () => {
 
   return [dayNow, timeTime];
 };
+
+export const getDayOfWeekNum = (dayOfWeekString) => {
+  switch (dayOfWeekString.toLowerCase()) {
+    case "sun":
+      return 0;
+    case "mon":
+      return 1;
+    case "tue":
+      return 2;
+    case "wed":
+      return 3;
+    case "thu":
+      return 4;
+    case "fri":
+      return 5;
+    case "sat":
+      return 6;
+    default:
+      return null;
+  }
+};
+
+export const getLastWeekdayInMonth = (year, month, dayOfWeek) => {
+  let date = new Date(year, month + 1, 0, 12);
+  date.setDate(date.getDate() - ((date.getDay() + 7 - dayOfWeek) % 7));
+  return date.toLocaleDateString("en-US", { month: "long", day: "numeric" });
+};
+
+export const getNthWeekdayInMonth = (day_of_week, week_of_month, date) => {
+  const weekday = getDayOfWeekNum(day_of_week);
+
+  if (week_of_month === -1) {
+    return getLastWeekdayInMonth(date.getFullYear(), date.getMonth(), weekday);
+  }
+
+  let nDate = new Date(date.getFullYear(), date.getMonth(), 1);
+  let add = ((weekday - nDate.getDay() + 7) % 7) + (week_of_month - 1) * 7;
+
+  nDate.setDate(1 + add);
+
+  return nDate;
+};
+
+export const getCurrentWeekOfMonth = () => {
+  const todaysDate = new Date();
+  const date = todaysDate.getDate();
+  const day = todaysDate.getDay();
+  const weekOfMonth = Math.ceil((date - 1 - day) / 7);
+
+  return weekOfMonth;
+};
+
+export const isLastWeekOfMonth = () => {
+  const todaysDate = new Date();
+  const currentMonth = todaysDate.getMonth();
+  const nextMonth = new Date(todaysDate);
+  nextMonth.setMonth(currentMonth + 1);
+  nextMonth.setDate(0);
+
+  const lastWeekStartDate = nextMonth.getDate() - 6;
+  const currentDate = todaysDate.getDate();
+
+  if (currentDate >= lastWeekStartDate) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+export const addMonths = (date, months) => {
```

Comment:
nit: if addMonths only used in the helper file, no need to `export`

Reply:
fixed